Sources/Defaults/Defaults+iCloud.swift

```diff
@@ -0,0 +1,571 @@
+		case .local:
+			guard let timestamp else {
+				return
+			}
+			key.suite.set(timestamp, forKey: "\(key.name)\(defaultsSyncKey)")
+		}
+	}
+
+	/**
+	Determine which data source has the latest data available by comparing the timestamps of the local and remote sources.
+	*/
+	private func latestDataSource(forKey key: Defaults.Keys) -> Defaults.DataSource {
+		// If the remote timestamp does not exist, use the local timestamp as the latest data source.
+		guard let remoteTimestamp = self.timestamp(forKey: key, .remote) else {
+			return .local
+		}
+		guard let localTimestamp = self.timestamp(forKey: key, .local) else {
+			return .remote
+		}
+
+		return localTimestamp > remoteTimestamp ? .local : .remote
+	}
+}
+
+/**
+`iCloudSynchronizer` notification related functions.
+*/
+extension iCloudSynchronizer {
+	private func registerNotifications() {
+		// TODO: Replace it with async stream when Swift supports custom executors.
+		NotificationCenter.default
+			.publisher(for: NSUbiquitousKeyValueStore.didChangeExternallyNotification)
+			.sink { [weak self] notification in
+				guard let self else {
+					return
+				}
+
+				self.didChangeExternally(notification: notification)
+			}
+			.store(in: &cancellables)
+
+		// TODO: Replace it with async stream when Swift supports custom executors.
+		#if os(iOS) || os(tvOS)
+		NotificationCenter.default
+			.publisher(for: UIScene.willEnterForegroundNotification)
+		#elseif os(watchOS)
+		NotificationCenter.default
+			.publisher(for: WKExtension.applicationWillEnterForegroundNotification)
+		#endif
+		#if os(iOS) || os(tvOS) || os(watchOS)
+			.sink { [weak self] notification in
+				guard let self else {
+					return
+				}
+
+				self.willEnterForeground(notification: notification)
+			}
+			.store(in: cancellables)
+		#endif
+	}
+
+	private func willEnterForeground(notification: Notification) {
+		remoteStorage.synchronize()
+	}
+
+	private func didChangeExternally(notification: Notification) {
+		guard notification.name == NSUbiquitousKeyValueStore.didChangeExternallyNotification else {
+			return
+		}
+
+		guard
+			let userInfo = notification.userInfo,
+			let changedKeys = userInfo[NSUbiquitousKeyValueStoreChangedKeysKey] as? [String],
+			// If `@TaskLocal timestamp` is not nil, it indicates that this notification is triggered by `syncRemoteStorageOnChange`, and therefore, we can skip updating the local storage.
+			Self.timestamp._defaults_isNil
+		else {
+			return
+		}
+
+		for key in self.keys where changedKeys.contains(key.name) {
+			guard let remoteTimestamp = self.timestamp(forKey: key, .remote) else {
+				continue
+			}
+			if
+				let localTimestamp = self.timestamp(forKey: key, .local),
+				localTimestamp >= remoteTimestamp
+			{
+				continue
+			}
+
+			self.enqueue {
+				await self.syncKey(key: key, .remote)
+			}
+		}
+	}
+}
+
+/**
+`iCloudSynchronizer` logging related functions.
+*/
+extension iCloudSynchronizer {
+	@available(macOS 11, iOS 14, tvOS 14, watchOS 7, *)
```

Comment:
```suggestion
	@available(macOS 11, iOS 14, tvOS 14, watchOS 7, visionOS 1, *)
```

Probably more places too.